### PR TITLE
fix(selene): diagnostics from json

### PR DESCRIPTION
**Pull request by mosheavni**
The pattern diagnostics was not working.
Instead of wasting time on patterns, there's an option to get a json response
the issue with that is that each diagnostic is in a json object on its own line, therefore I need to parse it row, decode and return the `from_json` helper.

Bottom line, before this PR, selene diagnostics are not working, after, they are.

### CODE REVIEW

**Comment by mochaaP** [comment]
please fix the tests, thanks 😄

**Comment by barrett-ruth** [comment]
I also prefer more systematic ways of parsing then patterns - however, how is your selene builtin "broken" (could I get an example)? Mine works just fine with `--version 0.7.2`:

image

File: lua/null-ls/builtins/formatting/npm_groovy_lint.lua [hunk 1]
```diff
@@ -13,7 +13,8 @@ return h.make_builtin({
     filetypes = { "groovy", "java", "Jenkinsfile" },
     generator_opts = {
         command = "npm-groovy-lint",
-        args = { "--format", "-" },
+        args = { "--failon", "none", "--format", "-" },
```
Comment (barrett-ruth) [inline lua/null-ls/builtins/formatting/npm_groovy_lint.lua]: This is fine but please kept PR changes atomic if you can @mosheavni

Comment (mosheavni) [reply lua/null-ls/builtins/formatting/npm_groovy_lint.lua]: huh? it's not on this PR

**Comment by mosheavni** [comment]
> please fix the tests, thanks 😄

Gladly, I honestly don't know how and spent too much time on it as-is. Mind giving me some hints here?

> I also prefer more systematic ways of parsing then patterns - however, how is your selene builtin "broken" (could I get an example)? Mine works just fine with `--version 0.7.2`:
> 
> image

I use selene 0.27.1 (installed by mason)
so maybe that's where the difference. I tried changing the pattern and then noticed we could get json output which is better.

**Comment by mosheavni** [comment]
Any ideas for this? @mochaaP 

https://github.com/nvimtools/none-ls.nvim/actions/runs/12485305514/job/34843919800?pr=222#step:5:525

```
Fail	||	diagnostics selene should create a diagnostic with an Error severity	
            vim/shared.lua:0: s: expected string, got table
            
            stack traceback:
            	vim/shared.lua: in function 'validate'
            	vim/shared.lua: in function 'gsplit'
            	vim/shared.lua: in function 'split'
            	...one-ls.nvim//lua/null-ls/builtins/diagnostics/selene.lua:24: in function 'parser'
            	...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:427: in function <...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:422>
            	
Fail	||	diagnostics selene should create a diagnostic	
            vim/shared.lua:0: s: expected string, got table
            
            stack traceback:
            	vim/shared.lua: in function 'validate'
            	vim/shared.lua: in function 'gsplit'
            	vim/shared.lua: in function 'split'
            	...one-ls.nvim//lua/null-ls/builtins/diagnostics/selene.lua:24: in function 'parser'
            	...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:446: in function <...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:441>
            	
Fail	||	diagnostics selene should not create a diagnostic for summary	
            vim/shared.lua:0: s: expected string, got table
            
            stack traceback:
            	vim/shared.lua: in function 'validate'
            	vim/shared.lua: in function 'gsplit'
            	vim/shared.lua: in function 'split'
            	...one-ls.nvim//lua/null-ls/builtins/diagnostics/selene.lua:24: in function 'parser'
            	...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:465: in function <...vim/none-ls.nvim/test/spec/builtins/diagnostics_spec.lua:460>
```

**Comment by mosheavni** [comment]
FIXED THE TESTS OMG i'm excited.

**Review by mochaaP — APPROVED** [review]
seems ok. tag me if you're ready for merge

**Comment by mosheavni** [comment]
@mochaaP  ready to merge :)

**Pull request merged**